packages/sdk/src/polkadot/PolkadotService.ts

```diff
@@ -65,7 +64,7 @@ export class PolkadotService {
       throw new Error('No native symbol key found');
     }
 
-    const asset = assetsMap.get(key);
+    const asset = this.chain.assetsData.get(key)?.asset;
```

Comment:
To remove the `assetsMap` dependency is correct but the change itself is not. As for `assetsData` in a chain, I meant to use it only if I need to overwrite something default for an asset. So it means that a chain may have not all assets in `assetsData` that a user can send. We need to find another solution.

Reply:
@ekenigs I see, alright. We can still pass ConfigService to PolkadotService.. We already do pass all opts from SDK down to getTransferData. WDYT ?

Reply:
Yes, I think it's a good idea.

Reply:
i'll push the fix this week :) was bit busy

Reply:
@ekenigs Pushed the fix. Sorry for delay, was busy with other work, finally got to xcm.

Reply:
@nohaapav no problem, I can understand that, I'm also very busy with other projects right now. Your changes look good. Please add the changeset and I will approve.